### DISCUSSION TITLE
value for multipleOf

### DISCUSSION THREAD

**Discussion by AleksanderBialik** [victools/jsonschema-generator#364] [Q&A] [answered]
Hey!
Is there a way to properly format the value for multipleOf for @Schema(multipleOf = 0.01)?
It generates schema with "multipleOf" : 0.0100000000000000055511151231257827021181583404541015625

**Comment by CarstenWickner** [comment, marked as answer]
Hi @AleksanderBialik,

Generally, yes. In the schema generator, the `multipleOf` value is treated as a `BigDecimal`, i.e., you can explicitly set the `scale` (number of decimal digits).
Unfortunately, on the swagger `@Schema` annotation, the `multipleOf` parameter is of type `double` which has this precision problem.

Short-term, you could basically re-implement the configuration with your desired precision:
https://github.com/victools/jsonschema-generator/blob/baea9b68690716a56afea2b5f46474296629a1eb/jsonschema-module-swagger-2/src/main/java/com/github/victools/jsonschema/module/swagger2/Swagger2Module.java#L92
https://github.com/victools/jsonschema-generator/blob/baea9b68690716a56afea2b5f46474296629a1eb/jsonschema-module-swagger-2/src/main/java/com/github/victools/jsonschema/module/swagger2/Swagger2Module.java#L304-L308

Mid-term, we could assume a certain maximum precision (let's say: rounded to 10 decimal digits) and then cut off trailing zeros. That would cover your use case at least.

**Reply by AleksanderBialik** [reply]
Hi @CarstenWickner,
Thanks for reply!
Actually I just did that what you recommended with re-implementing the `resolveMultipleOf` method and used `BigDecimal::valueOf`. This solution persists the precision set in the @Schema annotation.
Could the `BigDecimal::valueOf` actually replace the current solution in the `Swagger2Module.java`?

**Reply by CarstenWickner** [reply]
I didn't know `BigDecimal::valueOf` behaves differently in that way.
That is a great observation. I'll adapt that immediately in #365.
